packages/modules/src/firebase/functions-checkout/fix-items.ts

```diff
@@ -0,0 +1,219 @@
+            const flags: string[] = [];
+            item.flags.forEach((flag) => {
+              if (!flags.includes(flag)) {
+                flags.push(flag);
+              }
+            });
+            item.flags = flags;
+          }
+          //
+          // item.final_price = getPrice(body);
+          proceedItem();
+        }
+      }
+    };
+
+    const checkKitProduct = (kitProduct: Products, kitProductId: string) => {
+      if (item.kit_product) {
+        if (kitProduct.available && kitProduct.kit_composition) {
+          // check kit composition and quantities
+          let packQuantity = 0;
+          let isFixedQuantity = true;
+          let kitItem: Exclude<Products['kit_composition'], undefined>[number] | undefined;
+
+          kitProduct.kit_composition.forEach((currentKitItem) => {
+            if (currentKitItem.quantity) {
+              packQuantity += currentKitItem.quantity;
+            } else if (isFixedQuantity) {
+              isFixedQuantity = false;
+            }
+            if (currentKitItem._id === item.product_id) {
+              kitItem = currentKitItem;
+            }
+          });
+          if (!isFixedQuantity) {
+            // use parent product min quantity
+            packQuantity = kitProduct.min_quantity || 0;
+          }
+
+          if (kitItem && (kitItem.quantity === undefined
+            || item.quantity % kitItem.quantity === 0)) {
+            // valid kit item and quantity
+            let kitTotalQuantity = 0;
+            items.forEach((itemFind) => {
+              if (itemFind.kit_product && itemFind.kit_product._id === kitProductId) {
+                kitTotalQuantity += itemFind.quantity;
+              }
+            });
+
+            const minPacks = kitItem.quantity
+              ? item.quantity / kitItem.quantity
+              : 1;
+            if (kitTotalQuantity && kitTotalQuantity % (minPacks * packQuantity) === 0) {
+              // matched pack quantity
+            // item.kit_product.price = getPrice(kitProduct);
+              item.kit_product.pack_quantity = packQuantity;
+              if (item.kit_product.price) {
+                // set final price from kit
+                item.final_price = item.kit_product.price / packQuantity;
+              }
+              proceedItem();
+            }
+          }
+        }
+
+        // remove items with invalid kit
+        let index = 0;
+        while (index < items.length) {
+          const itemKit = items[index].kit_product;
+          if (itemKit && itemKit._id === kitProductId) {
+            items.splice(index, 1);
+          } else {
+            index += 1;
+          }
+        }
+        doFinally();
+      }
+    };
+
+    if (item.kit_product) {
+      // GET public kit product object
+      const kitProductId = item.kit_product._id;
+      // eslint-disable-next-line no-await-in-loop
+      const kitProduct = (await api.get(
+        `products/${kitProductId}`,
+      )).data;
```

Comment:
```suggestion
      const kitProduct = (await api.get(`products/${kitProductId}`, {
        isNoAuth: true,
      })).data;
```

Sempre melhor enviar requests sem autenticação quando você só busca dados públicos...